### PR TITLE
TNT-43107 Fix pageLoad not being mapped to target-global-mbox

### PR DESCRIPTION
**Pull request by XDex**
## Description

TNT-43107 Fix pageLoad not being mapped to target-global-mbox

## Related Issue

https://jira.corp.adobe.com/browse/TNT-43107

## How Has This Been Tested?

Adjusted unit test

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: packages/target-tools/src/utils.js [hunk 1]
```diff
@@ -44,6 +46,10 @@ function getNamesForRequested(itemsKey, deliveryRequest) {
       .forEach(item => {
         resultSet.add(item.name);
       });
+
+    if (isPojo(deliveryRequest[type].pageLoad)) {
```
Comment (jasonwaters) [inline packages/target-tools/src/utils.js:50]: `target-global-mbox` should never be added to a list of views, but it would be with this implementation

**Review by jasonwaters — CHANGES_REQUESTED** [review]
Hey Andrei,  I have a few concerns with this implementation:

1. I think`getMboxNames` should be true to it's name and only return mbox names, not the global mbox.  
2. There is a bug in the change you made to `getNamesForRequested`. It looks like `getViewNames` will include `target-global-mbox` in the list if `pageLoad` is in the request.  But `getViewNames` should only return an array of view names.

I see one of two paths to fix this bug:

1. Add the check for pageLoad in `hasRemoteDependency` and add `target-global-mbox` to the list of mboxes in that method only.  I don't think we ever want to see the global mbox in the list of mboxes except for this method anyway.  Leave the getMboxNames and getNamesForRequested methods untouched.
2. Expand the `getNamesForRequested` and `getMboxNames` method signatures to include a third param such as `includeGlobalMboxIfPageLoad = false`.  That way the default behavior is to not include `target-global-mbox` in the list of mbox names.  But it can be requested explicitly in the `hasRemoteDependency` method.

File: packages/target-tools/src/utils.js (hunk 1 above)
Comment (XDex) [reply packages/target-tools/src/utils.js]: Good catch, thanks! Fixed

**Comment by XDex** [comment]
@jasonwaters Thank you for the thorough review! Fixed as suggested in p.1, please take another look.

**Review by jasonwaters — APPROVED** [review]

**Pull request merged**